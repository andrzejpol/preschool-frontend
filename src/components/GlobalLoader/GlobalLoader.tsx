import React from 'react';
import {useSelector} from 'react-redux';
import './GlobalLoaderStyle.css';

const GlobalLoader = () => {
  const globalLoader = useSelector((state) => state.loaders.globalLoader);

  if (!globalLoader) return null;

  return (
    <div className="loader-overlay">
      <div className="spinner"></div>
    </div>
  );
};

export default GlobalLoader;
