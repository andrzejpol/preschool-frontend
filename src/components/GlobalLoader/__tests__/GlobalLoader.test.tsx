import React from "react";
import {render} from "@testing-library/react";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import GlobalLoader from "../GlobalLoader";

const mockStore = configureStore([]);

describe("GlobalLoader Component", () => {
  let store;

  it("matches the snapshot when loader is active", () => {
    store = mockStore({
      loaders: {globalLoader: true},
    });

    const {asFragment} = render(
      <Provider store={store}>
        <GlobalLoader/>
      </Provider>
    );

    // Snapshot assertion
    expect(asFragment()).toMatchSnapshot();
  });

  it("matches the snapshot when loader is inactive", () => {
    store = mockStore({
      loaders: {globalLoader: false},
    });

    const {asFragment} = render(
      <Provider store={store}>
        <GlobalLoader/>
      </Provider>
    );

    // Snapshot assertion
    expect(asFragment()).toMatchSnapshot();
  });
});
