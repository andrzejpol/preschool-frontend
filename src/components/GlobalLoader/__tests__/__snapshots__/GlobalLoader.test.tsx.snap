// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`GlobalLoader Component > matches the snapshot when loader is active 1`] = `
<DocumentFragment>
  <div
    class="loader-overlay"
  >
    <div
      class="spinner"
    />
  </div>
</DocumentFragment>
`;

exports[`GlobalLoader Component > matches the snapshot when loader is inactive 1`] = `<DocumentFragment />`;
